project/api/views.py

```diff
@@ -111,3 +113,23 @@ class NarrationViewSet(viewsets.ModelViewSet):
 
     queryset = Narration.objects.all()
     serializer_class = NarrationSerializer
+
+
+# https://medium.com/@mrgrantanderson/https-medium-com-serving-vector-tiles-from-django-38c705f677cf
+def mvt_tiles(request, zoom, x_cor, y_cor):
+    """
+    Custom view to serve Mapbox Vector Tiles for the custom polygon model.
+    """
+
+    with connection.cursor() as cursor:
+        cursor.execute(
+            (
+                "SELECT ST_AsMVT(tile) FROM (SELECT id, wikidata_id, date, "
+                "ST_AsMVTGeom(location, TileBBox(%s, %s, %s, 4326)) FROM api_cacheddata) AS tile"
```

Comment:
That is why we need to normalize rank values. 
Using this kind of filter is a right decision for regular MVT, however this will lead us to some limitations, e.g we will not be able to show events highlighted by narration on zoom levels, where event is not present.

Reply:
Sorry I'm a bit confused. What condition would we filter on for the `rank` and `zoom` parameters?
`we will not be able to show events highlighted by narration on zoom levels, where event is not present`
So to solve this we would increase the rank of these events before returning the MVT? How does normalization relate to this?

Reply:
> Sorry I'm a bit confused. What condition would we filter on for the rank and zoom parameters?

That is where normalization [normalization](https://en.wikipedia.org/wiki/Normalization_(statistics)) kicks in, we also need to decide where do we store additional field with result value  

> So to solve this we would increase the rank of these events before returning the MVT?

I am not sure we need to solve this. We have a fixed camera position in mind for each narration. We can perform filtering all events on the frontend but we still need the normalization rank.

cc @amaurymartiny @v1kun what do you think, should we filter events on backend or return all of them for now?

Reply:
After talking with @MiklerGM, we don't need normalization if we add the following:
- sort the events by ranking
- filter the top 20

We can hardcore both of these right now. This means that in each bounding box, the MVT will show the top 20 events in that box